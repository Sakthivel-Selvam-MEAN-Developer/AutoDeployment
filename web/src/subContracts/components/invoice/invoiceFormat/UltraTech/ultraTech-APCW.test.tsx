import { vi } from 'vitest'
import { render } from '@testing-library/react'
import UltraTechAPCW from './ultraTech-APCW'
import MockDate from 'mockdate'

vi.mock('to-words', () => ({
    ToWords: vi.fn().mockImplementation(() => ({
        convert: vi.fn().mockReturnValue('Ten Lakhs Only')
    }))
}))

describe('UltraTech_APCW invoice component', () => {
    test('renders correctly', () => {
        MockDate.set(new Date('2021-05-31'))
        const getProps = () => ({
            tripId: [
                {
                    overallTripId: 1,
                    tripId: 1,
                    tripName: 'LoadingToUnloading'
                }
            ],
            lastBillNumber: 'MGL23A-1',
            setLoading: vi.fn(),
            loading: false
        })

        const { asFragment } = render(<UltraTechAPCW {...getProps()} />)

        expect(asFragment()).toMatchSnapshot()
        MockDate.reset()
    })
})
