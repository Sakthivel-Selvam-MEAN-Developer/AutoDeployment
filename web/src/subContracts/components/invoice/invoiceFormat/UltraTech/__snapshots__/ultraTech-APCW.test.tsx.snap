// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UltraTech_APCW invoice component > renders correctly 1`] = `
<DocumentFragment>
  <main
    class="main"
    id="main"
  >
    <div
      class="header"
    >
      <p>
        CIN : 0
      </p>
      <p>
        TAX INVOICE
      </p>
      <div
        class="magnum-address"
      >
        <h1>
          MAGNUM LOGISTICS
        </h1>
        <p>
          201 BANGALOW THOTTAM,KADAPANALLUR POST,BHAVANI ERODE,ERODE,638311
        </p>
      </div>
      <div
        class="details"
      >
        <p>
          PAN : ASNAIIBSB
        </p>
        <p>
          GSTN : ASKJAHSKJHSJKS
        </p>
        <p>
          STATE : TAMILNADU
        </p>
        <p>
          STATE CODE : 33
        </p>
      </div>
      <p
        class="tax"
      >
        WHETHER TAX IS PAYBLE UNDER REVERSE CHARGE MECHANISIM : - NO
      </p>
    </div>
    <div
      class="title"
    >
      <p>
        Transportation Freight Bill
      </p>
    </div>
    <div
      class="place-of-supply"
    >
      <p>
        PLACE OF SUPPLY - Tamil Nadu
      </p>
      <p>
        HSN/SAC CODE : 91729
      </p>
    </div>
    <div
      class="address"
    >
      <div
        class="customer"
      >
        <div>
          <h4>
            M/s ULTRATECH CEMENT LIMITED
          </h4>
          <h4>
            Unit:- APCW
          </h4>
          <h4>
            PO:- BHOGASAMUDRAM, Tehsil : - TADIPATRI
          </h4>
          <h4>
            Dist:- ANANTAPUR
          </h4>
        </div>
        <div
          class="state-details"
        >
          <h4>
            State Name:- Andhra Pradesh (new)
          </h4>
          <h4>
            State code:- 37
          </h4>
          <h4>
            GSTN:- 37AAACL6442L1Z9
          </h4>
          <h4>
            PAN NO:- AAACL6442L
          </h4>
        </div>
      </div>
      <div
        class="vendor"
      >
        <div>
          <h4>
            Generation Date:- 21-02-2024 
          </h4>
          <h4>
            Vendor Code:- 0002406621 
          </h4>
          <h4>
            Vendor Name:- MAGNUM LOGISTICS
          </h4>
          <h2>
            Bill No:- MGL23A-267
          </h2>
        </div>
        <div
          class="flag"
        >
          <h4>
            FLAG:- Y 
          </h4>
          <h4>
             Bill Date:- 21-02-2024
          </h4>
        </div>
      </div>
    </div>
    <div
      class="table"
    >
      <table>
        <tbody>
          <tr>
            <th>
              SR No 
            </th>
            <th>
              DI no
            </th>
            <th>
              LR No
            </th>
            <th>
              Disp Date
            </th>
            <th>
              Sold to Party
            </th>
            <th>
              Destination
            </th>
            <th>
              Truck No
            </th>
            <th>
              Special Process Indicator
            </th>
            <th>
              Qty
            </th>
            <th>
              Rate
            </th>
            <th>
              Freight Amount
            </th>
            <th>
              Dedicated Freight
            </th>
            <th>
              SAP Lead
            </th>
            <th>
              Actual GPS Land
            </th>
            <th>
              Difference
            </th>
            <th>
              PTPK
            </th>
            <th>
              Final Freight
            </th>
          </tr>
          <tr>
            <td
              colspan="8"
              style="text-align: left;"
            >
              <h4>
                Total
              </h4>
            </td>
            <td>
              <b>
                0
              </b>
            </td>
            <td />
            <td>
              <b>
                0
              </b>
            </td>
            <td />
            <td />
            <td />
            <td />
            <td />
            <td>
              0
            </td>
          </tr>
          <tr>
            <td
              colspan="16"
              style="text-align: left;"
            >
              <h4>
                IGST 12.0%
              </h4>
            </td>
            <td>
              0
            </td>
          </tr>
          <tr>
            <td
              colspan="16"
              style="text-align: left;"
            >
              <h4>
                Grand Total
              </h4>
            </td>
            <td>
              0
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="amount-in-words"
    >
      <h4>
        Amount in words:- Ten Lakhs Only
      </h4>
    </div>
    <div
      class="pay-GST"
    >
      <h4>
        Person Liable to Pay GST : - MAGNUM LOGISTIC
      </h4>
    </div>
    <div
      class="declaration"
    >
      <h4>
        We hereby declare that though our aggregate turnover in any preceding financial year from 2017-18 onwards is more than the aggregate turnover notified under sub-rule (4) of rule 48, we are not required to prepare an invoice in terms of the provisions of the said sub-rule.
      </h4>
      <br />
      <h4>
        Declaration:- I/we have taken registration under the CGST Act, 2017 and have exercised the option to pay tax on services of GTA in relation to transport of goods supplied by us during the Financial Year 2023-2024 under forward charge.
      </h4>
    </div>
    <div
      class="digital-signature"
    >
      <div
        style="display: flex; align-items: baseline;"
      >
        <h5
          style="margin-right: 5px;"
        >
          For
        </h5>
        <h4>
          MAGNUM LOGISTICS
        </h4>
      </div>
      <img
        alt=""
        src="/src/subContracts/components/invoice/invoiceFormat/signature.png"
      />
      <h5>
        Authorized Signatory
      </h5>
    </div>
  </main>
</DocumentFragment>
`;
